data/core/logview.lua

```diff
@@ -122,12 +123,33 @@ function LogView:on_mouse_pressed(button, px, py, clicks)
 end
 
 
+function LogView:on_mouse_wheel(y, x)
+  if 0 < self.scroll.to.y then
+    if not self.user_scroll then
+      self.user_scroll = { x = self.scroll.to.x, y = self.scroll.to.y }
+    else
+      self.user_scroll.y = self.scroll.to.y
+    end
+  else
+    self.user_scroll = nil
+  end
+end
+
+
 function LogView:update()
   local item = core.log_items[#core.log_items]
   if self.last_item ~= item then
+    local lh = style.font:get_height() + style.padding.y
+    local diff_index = #core.log_items - self.last_item.index
```

Comment:
This can be a problem when the logs reach `config.max_log_items` and the table size remains the same.
I guess you could look for `self.last_item` every `update` to get `diff_index`.

Reply:
good point.
you mean loop through items backwards to find last one and evaluate the diff that way?

Reply:
Yep.

Reply:
maybe only start loop search if max_log_items has been reached, although there probably aren't many items added between updates. At least when logview is active, don't know if update is called while it is hidden. Haven't checked that.

Reply:
> although there probably aren't many items added between updates.

Yeah, most of the time it'll stop immediately.

> At least when logview is active, don't know if update is called while it is hidden. Haven't checked that.

It may be called (and we should probably look for *not* doing that), but I don't think it would impact performance in any measurable way.

Reply:
yes, may be better not to call update on hidden views.

side note: I set ``config.max_log_items`` to 10 but it clamped to 132. I guess the important part is that log-file doesn't grow :)